xchart/src/main/java/org/knowm/xchart/internal/chartpart/PlotContentLineChart.java

```diff
@@ -133,20 +143,23 @@ public void paint(Graphics2D g) {
       }
       Path2D.Double path = null;
 
-      while (xItr.hasNext()) {
-
-        double x = 0.0;
-        if (getChartPainter().getAxisPair().getXAxis().getAxisType() == AxisType.Number) {
-          x = ((Number) xItr.next()).doubleValue();
-          // System.out.println(x);
-        }
-        else if (getChartPainter().getAxisPair().getXAxis().getAxisType() == AxisType.Date) {
-          x = ((Date) xItr.next()).getTime();
-          // System.out.println(x);
-        }
-
-        if (getChartPainter().getStyleManager().isXAxisLogarithmic()) {
-          x = Math.log10(x);
+      double x = -0.5;
```

Comment:
Actually, after typing this out I got a new idea.  I added a second commit which fixes the issues with the graph point angles.  In combination with this hack, all I needed was to increase the xMax by one (so instead of count - 1, it is just count now).

Let me know your thoughts, is there a better way to do this?

Let me know if there is anything in this PR you think I should clean up.  I am still new to your code base, so I wont be surprised if there is better ways.